### ISSUE TITLE
Background2D crashes when both mask and coverage_mask are given

### ISSUE THREAD

**Issue by andresgur** [astropy/photutils#1224]
Here is my code 
```
fwhm_pixels = 20
source_mask = make_source_mask(image.data, nsigma=4,
                                       dilate_size=10, npixels=2 * int(fwhm_pixels), sigclip_sigma=5, sigclip_iters=5)
nx = image.wcs.naxis1
ny = image.wcs.naxis2
bkg_box_size = (nx // 4, ny // 4)
coverage_mask = (image.data == 0)`
bkg = Background2D(image.data, bkg_box_size, mask=source_mask, sigma_clip=sigma_clipping, exclude_percentile=10, coverage_mask=coverage_mask)
```
and the error:
~/anaconda3/lib/python3.8/site-packages/photutils/background/background_2d.py in __init__(self, data, box_size, mask, coverage_mask, fill_value, exclude_percentile, filter_size, filter_threshold, edge_method, sigma_clip, bkg_estimator, bkgrms_estimator, interpolator)
    370 
    371         self._prepare_data()
--> 372         self._calc_bkg_bkgrms()
    373 
    374     def _combine_masks(self):

~/anaconda3/lib/python3.8/site-packages/photutils/background/background_2d.py in _calc_bkg_bkgrms(self)
    731             bkgrms = self._make_2d_array(self._bkgrms1d)
    732         else:
--> 733             bkg = self._interpolate_meshes(self._bkg1d)
    734             bkgrms = self._interpolate_meshes(self._bkgrms1d)
    735 

~/anaconda3/lib/python3.8/site-packages/photutils/background/background_2d.py in _interpolate_meshes(self, data, n_neighbors, eps, power, reg)
    625         coords = np.array(list(product(range(self.nyboxes),
    626                                        range(self.nxboxes))))
--> 627         interp_func = ShepardIDWInterpolator(yx, data)
    628         img1d = interp_func(coords, n_neighbors=n_neighbors, power=power,
    629                             eps=eps, reg=reg)

~/anaconda3/lib/python3.8/site-packages/photutils/utils/interpolation.py in __init__(self, coordinates, values, weights, leafsize)
    133 
    134         if values.shape[0] != ncoords:
--> 135             raise ValueError('The number of values must match the number '
    136                              'of coordinates.')
    137 

ValueError: The number of values must match the number of coordinates

All source_mask, mask and image.data have the same shape. I believe the problem is that the data shape is being compared with the coordinates of the subboxes used in the Background computation. The code works fine if only mask or coverage_mask is given, the problem occurs only when both are given simultaneously.

**Comment by larrybradley** [comment]
@andresgur I tried with a dataset and cannot reproduce this error.  Would it be possible for you to provide your input `image.data` so I can investigate further?

**Comment by andresgur** [comment]
Thanks for looking into this @larrybradley . That's interesting...

I don't know how share that and the fits file it's rather big...

The image I'm using has the following id: hst_10354_01_acs_wfc_f550m and can be downloaded from https://mast.stsci.edu/portal/Mashup/Clients/Mast/Portal.html (look for NGC1672). Otherwise let me know how I can send it to you.

To read the image I use mpdaf package https://mpdaf.readthedocs.io/en/latest/ (simply `image = Image(file)`)

Thanks for having a look into this!

**Comment by larrybradley** [comment]
@andresgur - Many thanks for providing the dataset and reporting this bug!  I've tracked this down to a bug in `ShepardIDWInterpolator`.

Most of the pixels in your image are masked and as a result only 1 of the background meshes meets the threshold of having less than 10% masked pixels (`exclude_percentile=10`).  This triggered a bug in the interpolator having only 1 input data value.  With only 1 non-excluded mesh, the resulting background will be constant.  To improve this, you would need to increase `exclude_percentile` to allow more background meshes to contribute.

Also, I don't know your particular science case, but your current `source_mask` isn't aggressive enough if you want to mask the entire galaxy -- the background will be oversubtracted (subtracting the outskirts of the galaxy).  For your image, something like `source_mask = data > 0.05` gives better results.

**Comment by larrybradley** [comment]
This is fixed in #1226.
